code/modules/antagonists/clockcult/clock_mobs/clockwork_marauder.dm

```diff
@@ -124,3 +124,435 @@
+				amount *= 4 //if a wielded null rod is nearby, it takes four times the health damage
+				break
+	. = ..()
+	if(src && updating_health)
+		update_health_hud()
+		update_stats()
+
+/mob/living/simple_animal/hostile/clockwork/marauder/guardian/update_health_hud()
+	if(hud_used && hud_used.healths)
+		if(istype(hud_used, /datum/hud/marauder))
+			var/datum/hud/marauder/G = hud_used
+			var/resulthealth
+			if(host)
+				if(iscarbon(host))
+					resulthealth = "[round((abs(HEALTH_THRESHOLD_DEAD - host.health) / abs(HEALTH_THRESHOLD_DEAD - host.maxHealth)) * 100)]%"
+				else
+					resulthealth = "[round((host.health / host.maxHealth) * 100, 0.5)]%"
+			else
+				resulthealth = "NONE"
+			G.hosthealth.maptext = "<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='#AF0AAF'>HOST<br>[resulthealth]</font></div>"
+		hud_used.healths.maptext = "<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='#AF0AAF'>[round((health / maxHealth) * 100, 0.5)]%</font>"
+
+/mob/living/simple_animal/hostile/clockwork/marauder/guardian/proc/update_stats()
+	if(GLOB.ratvar_awakens)
+		speed = 0
+		melee_damage_lower = 20
+		melee_damage_upper = 20
+		attack_verb_continuous = "devastates"
+	else
+		var/healthpercent = (health/maxHealth) * 100
+		switch(healthpercent)
+			if(100 to 70) //Bonuses to speed and damage at high health
+				speed = 0
+				melee_damage_lower = 16
+				melee_damage_upper = 16
+				attack_verb_continuous = "viciously slashes"
+			if(70 to 40)
+				speed = initial(speed)
+				melee_damage_lower = initial(melee_damage_lower)
+				melee_damage_upper = initial(melee_damage_upper)
+				attack_verb_continuous = initial(attack_verb_continuous)
+			if(40 to 30) //Damage decrease, but not speed
+				speed = initial(speed)
+				melee_damage_lower = 10
+				melee_damage_upper = 10
+				attack_verb_continuous = "lightly slashes"
+			if(30 to 20) //Speed decrease
+				speed = 2
+				melee_damage_lower = 8
+				melee_damage_upper = 8
+				attack_verb_continuous = "lightly slashes"
+			if(20 to 10) //Massive speed decrease and weak melee attacks
+				speed = 3
+				melee_damage_lower = 6
+				melee_damage_upper = 6
+				attack_verb_continuous = "weakly slashes"
+			if(10 to 0) //We are super weak and going to die
+				speed = 4
+				melee_damage_lower = 4
+				melee_damage_upper = 4
+				attack_verb_continuous = "taps"
+
+//ATTACKING, BLOCKING, and COUNTERING
+
+/mob/living/simple_animal/hostile/clockwork/marauder/guardian/AttackingTarget()
+	if(is_in_host())
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/clockwork/marauder/guardian/bullet_act(obj/item/projectile/Proj)
+	if(blockOrCounter(null, Proj))
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/clockwork/marauder/guardian/hitby(atom/movable/AM, skipcatch, hitpush, blockedatom/movable/AM, datum/thrownthing/throwingdatum)
+	if(blockOrCounter(null, AM))
```

Comment:
it should be blocked not blockedatom/movable/AM
you already defined atom/movable/AM